src/crewai/agent.py

```diff
@@ -131,7 +131,7 @@ def check_agent_executor(self) -> "Agent":
     def execute_task(
         self,
         task: Any,
-        context: Optional[str] = None,
+        context: Optional[List[str]] = None,
```

Comment:
I just thought that the natural kind of context is a list of past messages in a conversation, and it's more natural to pass that through as a list instead of squashing it into a str

Reply:
I love it. It makes sense! My concern is the breaking change. What if we moved it to an `str | List[str] | None` type?